gcp/modules/couchdb/main.tf

```diff
@@ -142,15 +144,41 @@ resource "null_resource" "couchdb_enable_pv_backups" {
 }
 
 resource "null_resource" "couchdb_destroy_pvcs" {
+  count = "${var.execute_destroy_pvcs == "true" ? 1 : 0}"
+
   provisioner "local-exec" {
     when = "destroy"
 
     command = <<EOF
-      if [ "${var.execute_destroy_pvcs}" == "true" ]; then
-        for PVC in $(kubectl get pvc --namespace ${var.release_namespace} -o json | jq --raw-output '.items[] | select(.metadata.name | startswith("database-storage-couchdb")) | .metadata.name'); do
-          timeout -t 600 kubectl --namespace ${var.release_namespace} delete --ignore-not-found --grace-period=300 pvc $PVC
+      for PVC in "$(kubectl get pvc --namespace ${var.release_namespace} -o json | jq -r '.items[] | select(.metadata.name | startswith("database-storage-couchdb")) | .metadata.name')"; do
+        timeout -t 600 kubectl -n ${var.release_namespace} delete --ignore-not-found --grace-period=300 pvc $PVC
+      done
+      for PV in "$(kubectl get pv -o json | jq -r '.items[] | select (.spec.claimRef.name | startswith("database-storage-couchdb")) | select(.spec.claimRef.namespace == "${var.release_namespace}") | .metadata.name')"; do
+        timeout -t 600 kubectl delete --ignore-not-found --grace-period=300 pv $PV
+      done
+      # Some times GCP disks persist even after PVC and PVs destruction, removing them with gcloud to not leave stale resources behind
+      # More info: https://issues.gpii.net/browse/GPII-3743
+      RETRIES=10
+      RETRY_COUNT=1
+      while [ "$DISKS_DELETED" != "true" ]; do
+        DISKS_DELETED="true"
+        echo "[Try $RETRY_COUNT of $RETRIES] Deleting CouchDB disks..."
+        for DISK in $(gcloud compute disks list --filter description:couchdb --format json | jq -c '.[]'); do
+          echo "Trying to delete $(echo $DISK | jq -r .name)..."
+          timeout -t 60 gcloud compute disks delete $(echo $DISK | jq -r .selfLink) --quiet 2> /dev/null
+          if [ "$?" != "0" ]; then
+            DISKS_DELETED="false"
+          fi
         done
-      fi
+        if [ "$RETRY_COUNT" == "$RETRIES" ]; then
+          echo "Retry limit reached, giving up!"
+          exit 1
+        fi
+        if [ "$DISKS_DELETED" != "true" ]; then
+          sleep 10
+        fi
+        RETRY_COUNT=$(($RETRY_COUNT+1))
+      done
```

Comment:
Can we get rid of the `exit` clause and move all the loop control logic to the while clause? I think it less prone to errors and makes it more obvious what controls the loop. 

Smth. like:
```bash
while [ "$RETRY_COUNT" -le "$RETRIES" -a "DISKS_NOT_DELETED" != '0' ]; then
  DISKS_NOT_DELETED=0
  for DISK ...; do
    ..
    if [ "$?" != '0' ]; then
      DISKS_NOT_DELETED=$((DISKS_NOT_DELETED+1))
    fi
  done

  # Sleep only if we're not done nor last run    
  if [ "$RETRY_COUNT" -lt "$RETRIES" -a "DISKS_DELETED" != '0' ]; then
    echo "Failed to delete #{DISKS_NOT_DELETED} disks"
    sleep 10
  fi

  RETRY_COUNT=$((RETRY_COUNT+1))
done  
```

Reply:
But we would still need to exit with code 1 to fail the build if disks are not deleted and we run out of attempts. Added missing condition to retry check.

Reply:
I'll follow your advice and remove disk deletion code instead.